aten/src/ATen/native/vulkan/glsl/avg_pool2d.glsl

```diff
@@ -9,12 +9,20 @@ layout(std430) uniform;
 layout(set = 0, binding = 0, rgba16f) uniform PRECISION restrict writeonly image3D   uOutput;
 layout(set = 0, binding = 1)          uniform PRECISION                    sampler3D uInput;
 layout(set = 0, binding = 2)          uniform           restrict           Block {
-  ivec3 inputSize;
-  ivec3 outputSize;
-  ivec2 kernelSize;
-  ivec2 stride;
-  ivec2 padding;
-  ivec2 dilate;
+  int input_x;
+  int input_y;
+  int input_z;
```

Comment:
Just checked that this can be replaced with `ivec3 inputSize` without breaking it, but reverting back to `ivec3 outputSize` will cause the test to fail.

Reply:
I think it's got to do with alignment that Ivan mentioned. :) The first ivec3 starts off at 0, which is good, but _if_ it is an alignment issue, the second one is expected to start on a 16-byte boundary, which means we'll have to insert 4 bytes of unused padding.  This is an excellent find that shouldn't have been easy.  Thanks alot! :))  Now that we know where the problem is I'll spend some time thinking what the most elegant solution is.